File: backend-java/patient-service/src/main/java/com/cdm/patient/controller/ManagementSuggestionController.java
package com.cdm.patient.controller;

import com.cdm.common.security.IdentityPayload;
import com.cdm.patient.entity.ManagementSuggestionEntity;
import com.cdm.patient.service.ManagementSuggestionService;
import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.springframework.web.bind.annotation.*;
import java.util.Base64;
import java.util.List;

@RestController
@RequestMapping("/api/v1/management-suggestions")
public class ManagementSuggestionController {

    private final ManagementSuggestionService service;

    public ManagementSuggestionController(ManagementSuggestionService service) {
        this.service = service;
    }

    private IdentityPayload getIdentity(String base64Identity) {
        try {
            String json = new String(Base64.getDecoder().decode(base64Identity));
            return new ObjectMapper().readValue(json, IdentityPayload.class);
        } catch (Exception e) {
            throw new RuntimeException("Invalid identity header", e);
        }
    }

    @GetMapping("/{patientId}")
    public List<ManagementSuggestionEntity> getSuggestions(
            @RequestHeader("X-Identity-Base64") String identityHeader,
            @PathVariable Long patientId
    ) {
        IdentityPayload identity = getIdentity(identityHeader);
        return service.findAllForPatient(identity, patientId);
    }

    @PostMapping("/{patientId}")
    public ManagementSuggestionEntity createSuggestion(
            @RequestHeader("X-Identity-Base64") String identityHeader,
            @PathVariable Long patientId,
            @RequestParam String suggestionType,
            @RequestParam String content
    ) {
        IdentityPayload identity = getIdentity(identityHeader);
        return service.createSuggestion(identity, patientId, suggestionType, content);
    }
}

File: backend-nestjs/apps/patient/src/management-suggestion/management-suggestion.controller.ts
import { Controller } from '@nestjs/common';
import { MessagePattern, Payload } from '@nestjs/microservices';
import { ManagementSuggestionService } from './management-suggestion.service.js';
import type { PatientIdPayload, CreateSuggestionPayload } from '@cdm/shared';

@Controller()
export class ManagementSuggestionController {
  constructor(private readonly service: ManagementSuggestionService) {}

  @MessagePattern({ cmd: 'management_suggestion_find_all' })
  async findAll(@Payload() data: PatientIdPayload) {
    return this.service.findAllForPatient(data.identity, data.patientId);
  }

  @MessagePattern({ cmd: 'management_suggestion_create' })
  async createSuggestion(@Payload() data: CreateSuggestionPayload) {
    return this.service.createSuggestion(data.identity, data.patientId, { suggestionType: data.suggestionType, content: data.content });
  }
}
